### ISSUE TITLE
CRC non start in MAC M1 : Error Domain=VZErrorDomain Code=1

### ISSUE THREAD

**Issue by colussim** [crc-org/crc#3606]
### General information

  * OS: macOS Ventura 13.3.1
  * Hypervisor: vfkit
  * Did you run `crc setup` before starting it (Yes/No)? : Yes
  * Running CRC on: Laptop 

## CRC version
```bash
# Put `crc version` output here
CRC version: 2.17.0+44e15711
OpenShift version: 4.12.9
Podman version: 4.4.1
```
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
DEBU CRC version: 2.17.0+44e15711                 
DEBU OpenShift version: 4.12.9                    
DEBU Podman version: 4.4.1                        
DEBU Running 'crc status'                         
CRC VM:          Stopped
OpenShift:       Stopped (v4.12.9)
RAM Usage:       0B of 0B
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     73.68GB
Cache Directory: /Users/emmanuel.colussi/.crc/cache
DEBU Running 'sw_vers -productVersion'      
13.3.1
```

## CRC config
```bash
# Put `crc config view` output here
- consent-telemetry                     : yes
```

## Host Operating System
```bash
# put the output of `sw_vers` in case of Mac
ProductName:            macOS
ProductVersion:         13.3.1
BuildVersion:           22E261
```

### Steps to reproduce
```
crc start -p ~/Downloads/pull-secret.dms --log-level debug
DEBU CRC version: 2.17.0+44e15711                 
DEBU OpenShift version: 4.12.9                    
DEBU Podman version: 4.4.1                        
DEBU Running 'crc start'                          
DEBU Total memory of system is 34359738368 bytes  
DEBU Unable to find out if a new version is available: Get "https://access.cdn.redhat.com/content/origin/files/sha256/21/2146b0cffaeab3e039e1653f724007c1b414d0ffd2f28bc070adadf0ee185a24/release-info.json?_auth_=1681816019_b9f5236dbe94508c6e7b35724038f8da": context deadline exceeded (Client.Timeout exceeded while awaiting headers) 
DEBU Checking file: /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
INFO Checking if running as non-root              
INFO Checking if crc-admin-helper executable is cached 
DEBU Running '/Users/emmanuel.colussi/.crc/bin/crc-admin-helper-darwin --version' 
DEBU Found crc-admin-helper-darwin version 0.0.12 
DEBU crc-admin-helper executable already cached   
INFO Checking if running on a supported CPU architecture 
DEBU GOARCH is arm64 GOOS is darwin               
INFO Checking minimum RAM requirements            
DEBU Total memory of system is 34359738368 bytes  
INFO Checking if crc executable symlink exists    
INFO Checking if running emulated on a M1 CPU     
INFO Checking if vfkit is installed               
DEBU Running '/Users/emmanuel.colussi/.crc/bin/vfkit -v' 
DEBU Found vfkit version 0.0.4                    
INFO Checking if old launchd config for tray and/or daemon exists 
INFO Checking if crc daemon plist file is present and loaded 
DEBU Running 'bash -c launchctl list | grep com.redhat.crc.daemon | awk '{print $1}'' 
DEBU Checking file: /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
INFO Loading bundle: crc_vfkit_4.12.9_arm64...    
DEBU Checking file: /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
INFO Starting CRC VM for openshift 4.12.9...      
DEBU Updating CRC VM configuration                
DEBU kern.osproductversion is: 13.3.1             
DEBU Running /Users/emmanuel.colussi/.crc/bin/vfkit --cpus 4 --memory 9216 --kernel /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 --initrd /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img --kernel-cmdline console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a --device virtio-serial,logFilePath=/Users/emmanuel.colussi/.crc/machines/crc/vfkit.log --device virtio-fs,sharedDir=/Users/emmanuel.colussi,mountTag=dir0 --device virtio-rng --device virtio-blk,path=/Users/emmanuel.colussi/.crc/machines/crc/crc.img --device virtio-vsock,port=1024,socketURL=/Users/emmanuel.colussi/.crc/tap.sock,listen --timesync vsockPort=1234 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="&{4 9216 /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img vsockPort=1234 [virtio-serial,logFilePath=/Users/emmanuel.colussi/.crc/machines/crc/vfkit.log virtio-fs,sharedDir=/Users/emmanuel.colussi,mountTag=dir0 virtio-rng virtio-blk,path=/Users/emmanuel.colussi/.crc/machines/crc/crc.img virtio-vsock,port=1024,socketURL=/Users/emmanuel.colussi/.crc/tap.sock,listen]}" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="boot parameters:" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="\tkernel: /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="\tkernel command line:console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="\tinitrd: /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img" 
DEBU time="2023-04-18T12:57:03+02:00" level=info  
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="virtual machine parameters:" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="\tvCPUs: 4" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="\tmemory: 9216 MiB" 
DEBU time="2023-04-18T12:57:03+02:00" level=info  
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="Adding virtio-serial device (logFile: /Users/emmanuel.colussi/.crc/machines/crc/vfkit.log)" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="Adding virtio-fs device" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="Adding virtio-rng device" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="Adding virtio-blk device (imagePath: /Users/emmanuel.colussi/.crc/machines/crc/crc.img)" 
DEBU time="2023-04-18T12:57:03+02:00" level=info msg="Adding virtio-vsock device" 
DEBU Error: Error Domain=VZErrorDomain Code=1 Description="Internal Virtualization error. The virtual machine failed to start." UserInfo={ 
DEBU     NSLocalizedFailure = "Internal Virtualization error."; 
DEBU     NSLocalizedFailureReason = "The virtual machine failed to start."; 
DEBU }                                            
DEBU Usage:                                       
DEBU   vfkit [flags]                              
DEBU                                              
DEBU Flags:                                       
DEBU   -c, --cpus uint               number of virtual CPUs (default 1) 
DEBU   -d, --device stringArray      devices      
DEBU   -h, --help                    help for vfkit 
DEBU   -i, --initrd string           path to the virtual machine initrd 
DEBU   -k, --kernel string           path to the virtual machine linux kernel 
DEBU   -C, --kernel-cmdline string   linux kernel command line 
DEBU   -m, --memory uint             virtual machine RAM size in mibibytes (default 512) 
DEBU   -t, --timesync string         sync guest time when host wakes up from sleep 
DEBU   -v, --version                 version for vfkit 
DEBU                                              
DEBU Error Domain=VZErrorDomain Code=1 Description="Internal Virtualization error. The virtual machine failed to start." UserInfo={ 
DEBU     NSLocalizedFailure = "Internal Virtualization error."; 
DEBU     NSLocalizedFailureReason = "The virtual machine failed to start."; 
DEBU }                                            
DEBU Running 'sw_vers -productVersion'            
Error starting machine: Error in driver during machine start: exit status 1  
```

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
Deleted the instance

$ crc cleanup
INFO Unloading and removing the daemon plist file 
INFO Removing /etc/resolver/testing file          
INFO Stopping CRC vfkit process                   
INFO Removing CRC Specific entries from user's known_hosts file 
INFO Removing hosts file records added by CRC     
INFO Removing pull secret from the keyring        
INFO Removing older logs                          
INFO Removing CRC Machine Instance directory      
INFO Removing crc executable symlink              
Cleanup finished

$ crc setup
NFO Using bundle path /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64.crcbundle 
INFO Checking if running as non-root              
INFO Checking if crc-admin-helper executable is cached 
INFO Checking if running on a supported CPU architecture 
INFO Checking minimum RAM requirements            
INFO Checking if crc executable symlink exists    
INFO Creating symlink for crc executable          
INFO Checking if running emulated on a M1 CPU     
INFO Checking if vfkit is installed               
INFO Checking if CRC bundle is extracted in '$HOME/.crc' 
INFO Checking if /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64.crcbundle exists 
INFO Checking if old launchd config for tray and/or daemon exists 
INFO Checking if crc daemon plist file is present and loaded 
INFO Adding crc daemon plist file and loading it  
Your system is correctly setup for using CRC. Use 'crc start' to start the instance

$ crc start --log-level debug -p ~/Downloads/pull-secret.dms
DEBU CRC version: 2.17.0+44e15711                 
DEBU OpenShift version: 4.12.9                    
DEBU Podman version: 4.4.1                        
DEBU Running 'crc start'                          
DEBU Total memory of system is 34359738368 bytes  
DEBU No new version available. The latest version is 2.16.0 
DEBU Checking file: /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
INFO Checking if running as non-root              
INFO Checking if crc-admin-helper executable is cached 
DEBU Running '/Users/emmanuel.colussi/.crc/bin/crc-admin-helper-darwin --version' 
DEBU Found crc-admin-helper-darwin version 0.0.12 
DEBU crc-admin-helper executable already cached   
INFO Checking if running on a supported CPU architecture 
DEBU GOARCH is arm64 GOOS is darwin               
INFO Checking minimum RAM requirements            
DEBU Total memory of system is 34359738368 bytes  
INFO Checking if crc executable symlink exists    
INFO Checking if running emulated on a M1 CPU     
INFO Checking if vfkit is installed               
DEBU Running '/Users/emmanuel.colussi/.crc/bin/vfkit -v' 
DEBU Found vfkit version 0.0.4                    
INFO Checking if old launchd config for tray and/or daemon exists 
INFO Checking if crc daemon plist file is present and loaded 
DEBU Running 'bash -c launchctl list | grep com.redhat.crc.daemon | awk '{print $1}'' 
DEBU Checking file: /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
INFO Loading bundle: crc_vfkit_4.12.9_arm64...    
DEBU Using secret from configuration              
INFO Creating CRC VM for OpenShift 4.12.9...      
DEBU Running pre-create checks...                 
DEBU Creating machine...                          
DEBU /Users/emmanuel.colussi/.crc/machines/crc/crc.img is already 33285996544 bytes 
INFO Generating new SSH key pair...               
INFO Generating new password for the kubeadmin user 
DEBU Created /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
DEBU Machine successfully created                 
DEBU Checking file: /Users/emmanuel.colussi/.crc/machines/crc/.crc-exist 
INFO Starting CRC VM for openshift 4.12.9...      
DEBU Updating CRC VM configuration                
DEBU kern.osproductversion is: 13.3.1             
DEBU Running /Users/emmanuel.colussi/.crc/bin/vfkit --cpus 4 --memory 9216 --kernel /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 --initrd /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img --kernel-cmdline console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a --device virtio-serial,logFilePath=/Users/emmanuel.colussi/.crc/machines/crc/vfkit.log --device virtio-fs,sharedDir=/Users/emmanuel.colussi,mountTag=dir0 --device virtio-rng --device virtio-blk,path=/Users/emmanuel.colussi/.crc/machines/crc/crc.img --device virtio-vsock,port=1024,socketURL=/Users/emmanuel.colussi/.crc/tap.sock,listen --timesync vsockPort=1234 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="&{4 9216 /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img vsockPort=1234 [virtio-serial,logFilePath=/Users/emmanuel.colussi/.crc/machines/crc/vfkit.log virtio-fs,sharedDir=/Users/emmanuel.colussi,mountTag=dir0 virtio-rng virtio-blk,path=/Users/emmanuel.colussi/.crc/machines/crc/crc.img virtio-vsock,port=1024,socketURL=/Users/emmanuel.colussi/.crc/tap.sock,listen]}" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="boot parameters:" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="\tkernel: /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="\tkernel command line:console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="\tinitrd: /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img" 
DEBU time="2023-04-18T13:07:35+02:00" level=info  
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="virtual machine parameters:" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="\tvCPUs: 4" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="\tmemory: 9216 MiB" 
DEBU time="2023-04-18T13:07:35+02:00" level=info  
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="Adding virtio-serial device (logFile: /Users/emmanuel.colussi/.crc/machines/crc/vfkit.log)" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="Adding virtio-fs device" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="Adding virtio-rng device" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="Adding virtio-blk device (imagePath: /Users/emmanuel.colussi/.crc/machines/crc/crc.img)" 
DEBU time="2023-04-18T13:07:35+02:00" level=info msg="Adding virtio-vsock device" 
DEBU Error: Error Domain=VZErrorDomain Code=1 Description="Internal Virtualization error. The virtual machine failed to start." UserInfo={ 
DEBU     NSLocalizedFailure = "Internal Virtualization error."; 
DEBU     NSLocalizedFailureReason = "The virtual machine failed to start."; 
DEBU }                                            
DEBU Usage:                                       
DEBU   vfkit [flags]                              
DEBU                                              
DEBU Flags:                                       
DEBU   -c, --cpus uint               number of virtual CPUs (default 1) 
DEBU   -d, --device stringArray      devices      
DEBU   -h, --help                    help for vfkit 
DEBU   -i, --initrd string           path to the virtual machine initrd 
DEBU   -k, --kernel string           path to the virtual machine linux kernel 
DEBU   -C, --kernel-cmdline string   linux kernel command line 
DEBU   -m, --memory uint             virtual machine RAM size in mibibytes (default 512) 
DEBU   -t, --timesync string         sync guest time when host wakes up from sleep 
DEBU   -v, --version                 version for vfkit 
DEBU                                              
DEBU Error Domain=VZErrorDomain Code=1 Description="Internal Virtualization error. The virtual machine failed to start." UserInfo={ 
DEBU     NSLocalizedFailure = "Internal Virtualization error."; 
DEBU     NSLocalizedFailureReason = "The virtual machine failed to start."; 
DEBU }                                            
DEBU Running 'sw_vers -productVersion'            
Error starting machine: Error in driver during machine start: exit status 1
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

**Comment by cfergeau** [comment]
By any chance, are there any logs in `~/.crc/machines/crc/vfkit.log`?
Do you also get an error if you try to run this:
```
~/.crc/bin/vfkit --cpus 4 --memory 9216 --kernel /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 --initrd /Users/emmanuel.colussi/.crc/cache/crc_vfkit_4.12.9_arm64/initramfs-5.14.0-70.50.2.el9_0.aarch64.img --kernel-cmdline "console=hvc0 BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/vmlinuz-5.14.0-70.50.2.el9_0.aarch64 ignition.platform.id=qemu ostree=/ostree/boot.0/rhcos/b667cf58b05ef31fb1e9cc54f1c4f5a5f074df9b03d1a28846eabf17ccd643f4/0 root=UUID=8aa49d9c-6294-441d-b53d-b86cbd166df8 rw rootflags=prjquota boot=UUID=256a91e6-91a0-4fe7-b6ba-5d027f1ce56a" --device virtio-serial,logFilePath=/Users/emmanuel.colussi/.crc/machines/crc/vfkit-test.log --device virtio-rng --device virtio-blk,path=/Users/emmanuel.colussi/.crc/machines/crc/crc.img
```

**Comment by colussim** [comment]
the vfkit.log file is empty and same error when running directly the command...

**Comment by cfergeau** [comment]
> `CRC version: 2.17.0+44e15711`

Is this a build from git? I tested both release 2.16.0 and latest git on my M1 running 13.3.1, and I did not see this issue :-/

**Comment by colussim** [comment]
I had installed version 2.16 from the RedHat site and when I ran the crc setup command it downloaded 2.17 and the image crc_vfkit_4.12.9_arm64.crcbundle.Would you advise me to start from the git repository?

**Comment by cfergeau** [comment]
To be honest I don't understand how you ended up with 2.17, it's not released yet, and as far as I know we don't do such automatic upgrades. 2.16 should be good enough for now.

**Comment by praveenkumar** [comment]
@colussim Can you try to use latest version of OpenShift local and create a new issue if that still exist?
